docs/source/en/api/pipelines/stable_diffusion/adapter.mdx

```diff
@@ -0,0 +1,194 @@
+|[RzZ/sd-v1.4-adapter-seg](https://huggingface.co/RzZ/sd-v1-4-adapter-seg)<br/>*Trained with semantic segmentation*  | An [custom](https://github.com/TencentARC/T2I-Adapter/discussions/25) segmentation protocol image.|<a href="https://huggingface.co/RzZ/sd-v1-4-adapter-seg/resolve/main/sample_input.png"><img width="64" src="https://huggingface.co/RzZ/sd-v1-4-adapter-seg/resolve/main/sample_input.png"/></a>|<a href="https://huggingface.co/RzZ/sd-v1-4-adapter-seg/resolve/main/sample_output.png"><img width="64" src="https://huggingface.co/RzZ/sd-v1-4-adapter-seg/resolve/main/sample_output.png"/></a> |
+
+## Mix and match multiple adapters
+
+[`StableDiffusionAdapterPipeline`] also support using multiple type of *control image* at once with combination with [`MultiAdapter`].
+Here is a example of using keypose adapter for character posture control and depth adapter for outlining background.
+
+Just like the previous example, we will first prepare the *control image* for inference. One big difference when using [`MultiAdapter`] is that the *control image* we will send to pipeline is 
+combined from multiple images. In this example we stack two 3 channels RGB image(`cond_keypose`, `cond_depth`) together to create a 6 channels image tensor(`cond`).
+
+```py
+import torch
+from PIL import Image
+from diffusers.utils import load_image
+
+cond_keypose = load_image(
+    "https://huggingface.co/RzZ/sd-v1-4-adapter-keypose-depth/resolve/main/sample_input_keypose.png"
+)
+cond_depth = load_image("https://huggingface.co/RzZ/sd-v1-4-adapter-keypose-depth/resolve/main/sample_input_depth.png")
+cond = [[cond_keypose, cond_depth]]
+
+prompt = ["A man waling in an office room with nice view"]
+```
+
+Two *control image* should look like follows:
+
+![img](https://huggingface.co/RzZ/sd-v1-4-adapter-keypose-depth/resolve/main/sample_input_keypose.png)
+![img](https://huggingface.co/RzZ/sd-v1-4-adapter-keypose-depth/resolve/main/sample_input_depth.png)
+
+
+Now we can using `from_adapters` method combine keypose and depth adapter into one, then pass our newly created [`MultiAdapter`] to 
+[`StableDiffusionAdapterPipeline`]. You can also play around the value of `adapter_weights` to balance the control between adapters.
+
+```py
+from diffusers import StableDiffusionAdapterPipeline, MultiAdapter
+
+adapters = MultiAdapter.from_adapters(
+    [
+        Adapter.from_pretrained("RzZ/sd-v1-4-adapter-keypose"),
+        Adapter.from_pretrained("RzZ/sd-v1-4-adapter-depth"),
+    ],
+    adapter_weights=[0.8, 0.8],
+)
+adapters = adapters.to(torch.float16)
+
+pipe = StableDiffusionAdapterPipeline.from_pretrained(
+    "RzZ/sd-v1-4-adapter-pipeline",
+    torch_dtype=torch.float16,
+    adapter=adapters,
+)
+
+images = pipe(prompt, cond)
```

Comment:
Nice!

Can we update this to pass the two separate adapters to the pipeline constructor. We should not recommend that the pipeline user construct the separate class directly.

It should look like our examples for combining multiple controlnets 


```py
controlnet = [
    ControlNetModel.from_pretrained("lllyasviel/sd-controlnet-openpose", torch_dtype=torch.float16),
    ControlNetModel.from_pretrained("lllyasviel/sd-controlnet-canny", torch_dtype=torch.float16),
]

pipe = StableDiffusionControlNetPipeline.from_pretrained(
    "runwayml/stable-diffusion-v1-5", controlnet=controlnet, torch_dtype=torch.float16
)
```

Reply:
Change the example to:
```py
adapters = [
    Adapter.from_pretrained("RzZ/sd-v1-4-adapter-keypose"),
    Adapter.from_pretrained("RzZ/sd-v1-4-adapter-depth"),
]
pipe = StableDiffusionAdapterPipeline.from_pretrained(
    "CompVis/stable-diffusion-v1-4",
    torch_dtype=torch.float16,
    adapter=adapters,
)
```